react/machines/addressStateMachine.ts

```diff
@@ -0,0 +1,50 @@
+import { Machine } from 'xstate'
+
+import type { AddressMachineEvents } from './types'
+
+const addressStateMachine = Machine<unknown, AddressMachineEvents>(
+  {
+    initial: 'editAddress',
+    states: {
+      editAddress: {
+        initial: 'editing',
+        states: {
+          editing: {
+            on: {
+              SUBMIT_EDIT_ADDRESS: 'submitting',
+              RESET_ADDRESS: {
+                actions: 'goToCreateAddress',
+              },
+            },
+          },
+          submitting: {
+            invoke: {
+              src: 'tryToUpdateCompleteAddress',
```

Comment:
I'm not quite sure how, but can you add the error handling for this as well? I blocked the requests during address completion an when clicking to move to the next step, I got "infinite loading state" plus this error ([which I suppose comes from here?](https://github.com/vtex-apps/checkout-shipping/pull/28/files?file-filters%5B%5D=.json&file-filters%5B%5D=.md&file-filters%5B%5D=.png&file-filters%5B%5D=.ts&file-filters%5B%5D=.tsx&file-filters%5B%5D=No+extension&file-filters%5B%5D=dotfile#diff-5990586d0bda5f8e77ab587356994b80d0c87d60a5b3159c5f8e524b40c6e295R48)):
> Missing onError handler for invocation 'tryToUpdateCompleteAddress', error was 'Error: Network error: Failed to fetch'. Stacktrace was 'Error: Network error: Failed to fetch

Reply:
we don't have a screen on figma for this error state right now, I think I could add a transition back to the `editing` state to avoid the infinite loop.

Reply:
we don't have a screen on figma for this error state right now, I think I could add a transition back to the `editing` state to avoid the infinite loop.

Reply:
lgtm

Reply:
done in 04be8e7